ext/nnstreamer/tensor_filter/tensor_filter_onnxruntime.cc

```diff
@@ -0,0 +1,564 @@
+  }
+
+  model_path = g_strdup (prop->model_files[0]);
+
+  assert (model_path != nullptr);
+
+  if (!g_file_test (prop->model_files[0], G_FILE_TEST_IS_REGULAR)) {
+    const std::string err_msg
+        = "Given file " + (std::string) prop->model_files[0] + " is not valid";
+    std::cerr << err_msg << std::endl;
+    cleanup ();
+    throw std::invalid_argument (err_msg);
+  }
+
+  /** Read a model */
+  env = Ort::Env (ORT_LOGGING_LEVEL_WARNING, "nnstreamer_onnxruntime");
+  session = Ort::Session (env, model_path, sessionOptions);
+
+  input_num_tensors = session.GetInputCount ();
+  if (input_num_tensors <= 0 || input_num_tensors > NNS_TENSOR_SIZE_LIMIT) {
+    cleanup ();
+    throw std::invalid_argument (
+        "The number of tensors required by the given model exceeds the nnstreamer tensor limit (" NNS_TENSOR_SIZE_LIMIT_STR
+        " by default).");
+  }
+  inputInfo.num_tensors = input_num_tensors;
+
+  output_num_tensors = session.GetOutputCount ();
+  if (input_num_tensors <= 0 || input_num_tensors > NNS_TENSOR_SIZE_LIMIT) {
+    cleanup ();
+    throw std::invalid_argument (
+        "The number of tensors required by the given model exceeds the nnstreamer tensor limit (" NNS_TENSOR_SIZE_LIMIT_STR
+        " by default).");
+  }
+  outputInfo.num_tensors = output_num_tensors;
+
+  // initialize input info
+  input_shapes.resize (inputInfo.num_tensors);
+  input_types.resize (inputInfo.num_tensors);
+
+  for (i = 0; i < inputInfo.num_tensors; i++) {
+
+    // Get input names
+    auto input_name = session.GetInputNameAllocated (i, allocator);
+    input_names_allocated_strings.push_back (std::move (input_name));
+    input_names.push_back (input_names_allocated_strings.back ().get ());
+
+    // Get input types
+    ONNXTensorElementDataType type
+        = session.GetInputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetElementType ();
+
+    if (getTensorType (type, &inputInfo.info[i].type)) {
+      cleanup ();
+      throw std::invalid_argument ("Failed to convert ONNXTensorElement intput data type");
+    }
+    input_types[i] = type;
+
+    // Get input shapes/dims
+    size_t num_dims
+        = session.GetInputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetDimensionsCount ();
+    input_shapes[i]
+        = session.GetInputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetShape ();
+
+    // free dimensions are treated as 1 if not overriden
+    for (int64_t &dim : input_shapes[i]) {
+      if (dim == -1) {
+        dim = 1;
+      }
+    }
+
+    if (getTensorDim (inputInfo.info[i].dimension, input_shapes[i], num_dims)) {
+      cleanup ();
+      throw std::invalid_argument ("Shape input rank too high.");
+    }
+
+    inputInfo.info[i].name = nullptr;
+
+    gchar *dim;
+    dim = gst_tensor_get_dimension_string (inputInfo.info[i].dimension);
+
+    nns_logd ("inputInfo[%zu] >> name[%s], type[%d], dim[%s]", i,
+        inputInfo.info[i].name, inputInfo.info[i].type, dim);
+    g_free (dim);
```

Comment:
```suggestion
    {
        g_autofree gchar *dim = gst_tensor_get_dimension_string (inputInfo.info[i].dimension);

        nns_logd ("inputInfo[%zu] >> name[%s], type[%d], dim[%s]", i,
            inputInfo.info[i].name, inputInfo.info[i].type, dim);
    }
```

Even though GLIB-2.0 provides features so-called "smart" ptr, it would be better to use C++ built-in features.

Reply:
Thanks. updated it.